### PR TITLE
PMM-4041 Implement inventory list.

### PR DESCRIPTION
**Pull request by BupycHuk**

### CODE REVIEW

**Review by idexter — APPROVED** [review]

File: commands/inventory/list_agents.go [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+	"strings"
+
+	"github.com/AlekSi/pointer"
+	"github.com/percona/pmm/api/inventorypb/json/client"
+	"github.com/percona/pmm/api/inventorypb/json/client/agents"
+
+	"github.com/percona/pmm-admin/commands"
+)
+
+var listAgentsGenericResultT = commands.ParseTemplate(`
```
Comment (AlekSi) [inline commands/inventory/list_agents.go]: Is that `Generic` part copied from Generic Node template? That's Node type, not just word.

File: commands/inventory/list_agents.go [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+type listResultAgent struct {
+	AgentType  string `json:"agent_type"`
+	AgentID    string `json:"agent_id"`
+	PMMAgentID string `json:"pmm_agent_id"`
+	ServiceID  string `json:"service_id"`
+	Status     string `json:"status"`
+}
+
+type listAgentsResult struct {
+	Agents []listResultAgent
```
Comment (AlekSi) [inline commands/inventory/list_agents.go]: Please add json tag. We need them for all output fields. Nodes and Service too.

**Review by AlekSi — CHANGES_REQUESTED** [review]

File: commands/inventory/list_agents.go (hunk 1 above)
Comment (BupycHuk) [reply commands/inventory/list_agents.go]: yeah, my fault

File: commands/inventory/list_nodes.go [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+{{ range .Nodes }}
+{{- printf "%-13s" .NodeType }} {{ printf "%-20s" .NodeName }} {{ printf "%-17s" .Address }} {{ .NodeID }}
+{{ end }}
+`)
+
+type listResultNode struct {
+	NodeType string
+	NodeName string
+	Address  string
+	NodeID   string
```
Comment (AlekSi) [inline commands/inventory/list_nodes.go]: JSON tags.

File: commands/inventory/list_nodes.go [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+			NodeType: "Generic",
+			NodeName: node.NodeName,
+			Address:  node.Address,
+			NodeID:   node.NodeID,
+		})
+	}
+
+	for _, node := range result.Payload.Container {
+		nodes = append(nodes, listResultNode{
+			NodeType: "Generic",
```
Comment (AlekSi) [inline commands/inventory/list_nodes.go]: That should be container.

**Comment by AlekSi** [comment]
Will fix now…

**Pull request merged**